test/system/211-dbus.bats

```diff
@@ -87,3 +87,351 @@ teardown() {
+
+  assert_success
+  assert_line --index 0 "$expected_response"
+
+  if check_bats_version 1.10.0; then
+    assert [ ${#lines[@]} -eq 1 ]
+  else
+    assert [ ${#lines[@]} -eq 2 ]
+  fi
+
+  # shellcheck disable=SC2154
+  assert [ ${#stderr_lines[@]} -eq 0 ]
+}
+
+@test "dbus: session bus inside RHEL 8.7" {
+  create_distro_container rhel 8.7 rhel-toolbox-8.7
+  expected_response="$(gdbus call \
+                         --session \
+                         --dest org.freedesktop.DBus \
+                         --object-path /org/freedesktop/DBus \
+                         --method org.freedesktop.DBus.Peer.Ping)"
+
+  run --keep-empty-lines --separate-stderr "$TOOLBOX" run --distro rhel --release 8.7 sh -c 'gdbus call \
+                                                                   --session \
+                                                                   --dest org.freedesktop.DBus \
+                                                                   --object-path /org/freedesktop/DBus \
+                                                                   --method org.freedesktop.DBus.Peer.Ping'
+
+  assert_success
+  assert_line --index 0 "$expected_response"
+
+  if check_bats_version 1.10.0; then
+    assert [ ${#lines[@]} -eq 1 ]
+  else
+    assert [ ${#lines[@]} -eq 2 ]
+  fi
+
+  # shellcheck disable=SC2154
+  assert [ ${#stderr_lines[@]} -eq 0 ]
+}
+
+@test "dbus: system bus inside RHEL 8.7" {
+  create_distro_container rhel 8.7 rhel-toolbox-8.7
+  expected_response="$(gdbus call \
+                         --system \
+                         --dest org.freedesktop.systemd1 \
+                         --object-path /org/freedesktop/systemd1 \
+                         --method org.freedesktop.DBus.Properties.Get org.freedesktop.systemd1.Manager Version)"
+
+  run --keep-empty-lines --separate-stderr "$TOOLBOX" run --distro rhel --release 8.7 sh -c 'gdbus call \
+                                                                   --system \
+                                                                   --dest org.freedesktop.systemd1 \
+                                                                   --object-path /org/freedesktop/systemd1 \
+                                                                   --method org.freedesktop.DBus.Properties.Get \
+                                                                     org.freedesktop.systemd1.Manager Version'
+
+  assert_success
+  assert_line --index 0 "$expected_response"
+
+  if check_bats_version 1.10.0; then
+    assert [ ${#lines[@]} -eq 1 ]
+  else
+    assert [ ${#lines[@]} -eq 2 ]
+  fi
+
+  # shellcheck disable=SC2154
+  assert [ ${#stderr_lines[@]} -eq 0 ]
+}
+
+@test "dbus: session bus inside Ubuntu 16.04" {
+  create_distro_container ubuntu 16.04 ubuntu-toolbox-16.04
+
+  run --keep-empty-lines --separate-stderr "$TOOLBOX" run --distro ubuntu --release 16.04 sh -c 'sudo apt-get update && sudo apt-get install -y libglib2.0-bin'
```

Comment:
It will be better if we can avoid using `apt-get`, because it's access of the network introduces another point of failure for the tests.  Sometimes, inevitably, the network will have problems and the tests will fail.

Did you try using [busctl(1)](https://www.freedesktop.org/software/systemd/man/busctl.html) as I had mentioned before in https://github.com/containers/toolbox/issues/1330 ?

Reply:
Yes, I tried but didn't manage to make it work. Then I found out this solution, so I did it this way. I can try to make it work with [busctl(1)](https://www.freedesktop.org/software/systemd/man/busctl.html)

Reply:
We definitely do not want to install any packages (or in general do anything that relies on an external network/resource) as that is inherently unreliable. Also, the added runtime cost per test is way too high.

The idea to use `busctl(1)` seems to be a nice way to go.